src/path_info.cpp

```diff
@@ -158,6 +158,12 @@ void PATH_INFO::set_standard_filenames(void)
     update_pathname("obsolete-mods", FILENAMES["moddir"] + "obsolete-mods.json");
     update_pathname("defaultsounddir", FILENAMES["datadir"] + "sound");
 
+    // User's data directories
+    update_pathname("user_datadir", FILENAMES["user_dir"] + "data/");
+    update_pathname("user_gfxdir", FILENAMES["user_datadir"] + "gfx/");
```

Comment:
I'm confused, because it looks to me like the gfx case is the broken one,
it will trigger the "external user dir" behavior even when the game is
deployed in the "run from unpacked archive" configuration since the paths
don't match, and all you need to do is define that path as FILENAMES
["user_dir"] + "gfx".

This difference makes me think we need to be more formal about checking
this, we need to write up the expected behavior of each configuration to
review the actual behavior against.

> Just curious, why even have the option
> to use the current directory for user_dir?

That's a hilarious question. That is the default, and the dominant way the
game is packaged and distributed. Literally everything else is support for
secondary packaging.
Look at the RELEASE target, it doesn't set any of those flags when building.